src/test/java/teammates/test/cases/webapi/GetStudentProfileActionTest.java

```diff
@@ -28,80 +31,240 @@ protected String getRequestMethod() {
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, classmate.getEmail(),
+        };
+        verifyCannotAccess(submissionParams);
+    }
 
-        ______TS("Cannot view another student's profile");
+    @Test
+    public void testAccessControl_studentAccessStudentInOtherCourse_shouldFail() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        StudentAttributes studentNotInCourse1 = typicalBundle.students.get("student1InCourse2");
+        loginAsStudent(student1InCourse1.googleId);
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, student1InCourse1.getCourse(),
+                Const.ParamsNames.STUDENT_EMAIL, studentNotInCourse1.getEmail(),
+        };
+        verifyCannotAccess(submissionParams);
+    }
 
-        AccountAttributes student1 = typicalBundle.accounts.get("student1InCourse1");
-        AccountAttributes student2 = typicalBundle.accounts.get("student2InCourse1");
-        loginAsStudent(student2.googleId);
+    @Test
+    public void testAccessControl_instructorAccessProfileWithMissingStudentEmail_shouldFail() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+        };
+        verifyInaccessibleForInstructors(submissionParams);
+    }
 
+    @Test
+    public void testAccessControl_instructorAccessProfileWithMissingCourseId_shouldFail() {
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
         String[] submissionParams = new String[] {
-                Const.ParamsNames.STUDENT_ID, student1.googleId,
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
         };
+        verifyInaccessibleForInstructors(submissionParams);
+    }
 
-        GetStudentProfileAction action = getAction(submissionParams);
-        assertThrows(UnauthorizedAccessException.class, () -> action.checkAccessControl());
+    @Test
+    public void testAccessControl_instructorAccessProfileFromHisCourse_shouldPass() {
+        InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
+        StudentAttributes student1InCourse1 = typicalBundle.students.get("student1InCourse1");
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
+                Const.ParamsNames.STUDENT_EMAIL, student1InCourse1.getEmail(),
+        };
+        verifyAccessibleForInstructorsOfTheSameCourse(submissionParams);
     }
 
+    @Test
+    public void testAccessControl_instructorWithoutViewStudentInSectionPrivilege_shouldFail() throws Exception {
```

Comment:
You should use `verifyInaccessibleWithoutViewStudentInSectionsPrivilege` in BaseActionTest?

Reply:
This is because I need to specify which privilege is needed to be tested, see comments below